dbcli/pgcli
# Modernize code to Python 3.6+

## Description

A series of independent commits to modernize the code to Python 3.6. Partially from `pyupgrade --py36-plus`, partially from PyCharm suggestions, partially manual find&replace. The individual commits contain very similar small diffs, so please process them commit after commit. If you prefer, I can split them into individual PRs.

1. `class A(object)` can be written as `class A:`
2. replace `dict([…])` and `set([…])` with `{…}`
3. use f-strings or compact `.format`
4. use `yield from` instead of `yield` in a `for` loop
5. import `mock` from `unittest`
6. expect `OSError` instead of `IOError` or `select` error
7. use Python3 defaults for file reading or `super()`
8. remove redundant parenthesis (keep those in tuples though)
9. shorten set intersection instead of creating lists
10. backslashes in strings do not have to be escaped if prepended with `r`

I would like to hear your opinion on different points and would be happy to adjust them to your needs.


## Checklist

- [ ] I've added this contribution to the `changelog.rst`.
- [ ] I've added my name to the `AUTHORS` file (or it's already there).

- [x] I installed pre-commit hooks (`pip install pre-commit && pre-commit install`), and ran `black` on my code.
- [x] Please squash merge this pull request (uncheck if you'd like us to merge as multiple commits)
